### PR TITLE
Enable support for Python 3.7

### PR DESCRIPTION
**Pull request by theunkn0wn1**
This PR enables support for Python 3.7 and resolves the conflict around `async`

# Major changes made
- Removed the `Pydle.async` wrapper library, see below
- wherever possible generator-based coroutines have been replaced with native coroutines (see [PEP 492](https://www.python.org/dev/peps/pep-0492/))
  - where that was not possible, the generator based coroutine was preserved. (Namely `BasicClient.whois`)
- Custom wrappers / implementations of `asyncio` methods and functions have been replaced with their `asyncio` equivalents
- Minimum language requirement raised to Python 3.5

## reasoning for throwing out `pydle.async`
There are two main reasons for removing this module.
The first being the [module's name is invalid as of python 3.7](https://docs.python.org/3.7/whatsnew/3.7.html) Therefore, there was one of two possible courses of action.

1. rename the module, refactor usages
2. Remove the module entirely, use `asyncio` methods directly

The first course of action adds technical debt, and the module is entirely superseded by `asyncio` methods. Further, this wrapper introduces what appears to be a needless level of abstraction and re-invention of the wheel, as it re-implements various core `asyncio` methods. 
Further, as Shizmob noted in #71 
>>> The asyncio branch is the future for pydle, so I don't think I'll be spending much work on making it work with Tornado 5.0+.

The second course of action also simplifies the code base, making it more maintainable, testable. and easier to read. In the immortal words of Tim Peters:
>Flat is better than nested.
Sparse is better than dense.
Readability counts.

## what this PR does not do
- this PR does not modify Pydle's public Interface and shouldn't effect functionality. Provided developers are using a recent version of python, and are not making use of `pydle.async`,  this PR should not break their things. 

- this PR does not resolve the errors spotted in `features/ctcp.py` nor the errors spotted in `features/ircv3/monitor.py` 
  - These will probably become Issue tickets and/or fixed in a later pull request

## Caveats:
as Pydle's unit tests are [currently broken](#78 ), I am unable to assert I broke nothing.
That said, i preformed some smoke and integration tests using both basic examples (single and pool) without raising errors, and it appears overall functionality is unchanged

### CODE REVIEW

**Comment by Shizmob** [comment]
Thanks so much for this effort, and sorry for responding so late! I'll take a closer look tomorrow, but a lot seems good. I'm still not super sure on the complete removal of `async.py` though, let me mull it over a little bit.

**Comment by theunkn0wn1** [comment]
If you would prefer, i can re-implement this PR with the rename-async-lib approach. I don't believe this is the best course of action, and it doesn't resolve the underlying issues with rolling our own event loop implementation / async wrapper library. Using the asyncio library directly simplifies the implementation, which in turn makes this code easier to test and maintain.

**Comment by wavenator** [comment]
Can you please approve this PR.
Supporting Python3.7 is crucial these days.

File: pydle/client.py [hunk 1]
```diff
@@ -36,15 +39,17 @@ class BasicClient:
...
+                 eventloop=None, **kwargs):
         """ Create a client. """
         self._nicknames = [nickname] + fallback_nicknames
         self.username = username or nickname.lower()
         self.realname = realname or nickname
         if eventloop:
             self.eventloop = eventloop
         else:
-            self.eventloop = async.EventLoop()
+            self.eventloop: BaseEventLoop = get_event_loop()
```
Comment (Harmon758) [inline pydle/client.py]: Note: This dropped support for Python <3.6

**Comment by Harmon758** [comment]
Note: Support for Python <3.6 was dropped with the [type annotation](https://github.com/Shizmob/pydle/pull/79/files#diff-d989e47e22fa23b1769d31ecaade09a5R51).

File: pydle/client.py (hunk 1 above)
Comment (Harmon758) [inline pydle/client.py]: Note: This dropped support for Python <3.6

File: pydle/client.py [hunk 2]
```diff
@@ -349,56 +343,54 @@ def _create_message(self, command, *params, **kwargs):
...
         """ Handle received data. """
         self._receive_buffer += data
 
         # Schedule new timeout event.
         if self._ping_checker_handle:
-            self.eventloop.unschedule(self._ping_checker_handle)
-        self._ping_checker_handle = self.eventloop.schedule_async_in(self.PING_TIMEOUT, self._perform_ping_timeout())
+            self._ping_checker_handle.cancel()
+
+        self._ping_checker_handle: TimerHandle = self.eventloop.call_later(self.PING_TIMEOUT,
```
Comment (Harmon758) [inline pydle/client.py]: Note: This dropped support for Python <3.6

File: pydle/client.py [hunk 3]
```diff
@@ -450,14 +441,17 @@ class ClientPool:
     """ A pool of clients that are ran and handled in parallel. """
 
     def __init__(self, clients=None, eventloop=None):
-        self.eventloop = eventloop or async.EventLoop()
-        self.clients = set(clients or [])
+        self.eventloop = eventloop if eventloop else new_event_loop()
+        self.clients: Set[BasicClient] = set(clients or [])
```
Comment (Harmon758) [inline pydle/client.py]: Note: This dropped support for Python <3.6

**Comment by Harmon758** [comment]
Note: Separately, Python 3.4 support was also dropped with `async def`/`await` syntax.

**Pull request merged**